loglist/diff_check.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package loglist
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"reflect"
+	"sort"
+
+	ct "github.com/google/certificate-transparency-go"
+)
+
+type warningList struct {
+	warnings []string
+}
+
+func (wl *warningList) addWarning(w string) {
+	if w != "" {
+		wl.warnings = append(wl.warnings, w)
+	}
+}
+
+// Check: operator IDs set of branch is equal to or wider than master one.
+// No restriction on description mismatches.
+func checkMasterOpsMatchBranch(master *LogList, branch *LogList, wl *warningList) {
+	var masterOps = master.GetOperatorIdSet()
+	var branchOps = branch.GetOperatorIdSet()
+	for opId, _ := range masterOps {
+		if branchOps[opId] == "" {
+			wl.addWarning(fmt.Sprintf(
+				"Operator %q id=%d present at master log list but missing at branch.",
+				masterOps[opId], opId))
+		}
+	}
+}
+
+// Check: 2 logs are functionally identical.
+func (log1 *Log) checkEquivalence(log2 *Log, wl *warningList) {
+	// Description and STH comparison are omitted.
+
+	if !bytes.Equal(log1.Key, log2.Key) {
+		wl.addWarning(fmt.Sprintf(
+			"Log %q and log %q have different keys.",
+			log1.Description, log2.Description))
+	}
+	if log1.MaximumMergeDelay != log2.MaximumMergeDelay {
+		wl.addWarning(fmt.Sprintf(
+			"Maximum merge delay mismatch for logs %q and %q: %d != %d.",
+			log1.Description, log2.Description, log1.MaximumMergeDelay, log2.MaximumMergeDelay))
+	}
+	// Strong assumption: operators IDs are semantically same across logs.
```

Comment:
This assumption also applies to `checkMasterOpsMatchBranch`.  @taknira , is it a valid assumption (in theory or in practice)?    Is it possible/allowed to renumber the operator ID numbers and consider the result the "same" log list?